source/views/sis/course-search/search.js

```diff
@@ -44,13 +49,17 @@ class CourseSearchView extends React.PureComponent<Props, State> {
 	}
 
 	state = {
+		dataLoading: true,
 		searchResults: [],
 		searchActive: false,
 		searchPerformed: false,
 	}
 
 	componentDidMount() {
-		this.props.updateCourseData()
+		this.props.loadCourseDataIntoMemory().then(() => {
```

Comment:
I think we need to make this callback async and add an `await` before `this.props.updateCourseData()`. Otherwise when no there is no data in the cache, `dataLoading` will be set to false right away and there will be no loading screen..I tested this and changing it to 
```javascript
this.props.loadCourseDataIntoMemory().then(async () => {
	await this.props.updateCourseData()
	this.setState(() => ({dataLoading: false}))
})
``` 
seems to do the trick.